pkg/cosign/tuf/client.go

```diff
@@ -129,101 +129,105 @@ func isExpiredMetadata(metadata []byte) bool {
-		}
+	local, err := tuf_leveldbstore.FileLocalStore(localCacheDBPath)
+	if err != nil {
+		return nil, errors.Wrap(err, "creating cached local store")
+	}
 
-		// We may need to download latest metadata and targets if the cache is un-initialized or expired.
-		trustedMeta, err := local.GetMeta()
-		if err != nil {
-			return nil, errors.Wrap(err, "getting trusted meta")
-		}
+	// We may need to download latest metadata and targets if the cache is un-initialized or expired.
+	trustedMeta, err := local.GetMeta()
+	if err != nil {
+		return nil, errors.Wrap(err, "getting trusted meta")
+	}
 
-		uninitializedClient := client.NewClient(local, remote)
-
-		trustedTimestamp, ok := trustedMeta["timestamp.json"]
-		if !ok || isExpiredMetadata(trustedTimestamp) {
-			var trustedRoot []byte
-			trustedRoot, ok := trustedMeta["root.json"]
-			if !ok {
-				// Use embedded root or altRoot as trusted if cached root does not exist
-				if altRoot != nil {
-					trustedRoot = altRoot
-				} else {
-					trustedRoot, err = embeddedReadFile("repository", "root.json")
-					if err != nil {
-						return nil, errors.Wrap(err, "reading embedded trusted root")
-					}
+	uninitializedClient := client.NewClient(local, remote)
+
+	trustedTimestamp, ok := trustedMeta["timestamp.json"]
+	if !ok || isExpiredMetadata(trustedTimestamp) {
+		var trustedRoot []byte
+		trustedRoot, ok := trustedMeta["root.json"]
+		if !ok {
+			// Use embedded root or altRoot as trusted if cached root does not exist
+			if altRoot != nil {
+				trustedRoot = altRoot
+			} else {
+				trustedRoot, err = embeddedReadFile("repository", "root.json")
+				if err != nil {
+					return nil, errors.Wrap(err, "reading embedded trusted root")
 				}
 			}
-			rootKeys, rootThreshold, err := getRootKeys(trustedRoot)
-			if err != nil {
-				return nil, errors.Wrap(err, "bad trusted root")
-			}
-
-			if err := uninitializedClient.Init(rootKeys, rootThreshold); err != nil {
-				return nil, errors.Wrap(err, "initializing root client")
-			}
-			if err := updateMetadataAndDownloadTargets(uninitializedClient); err != nil {
-				return nil, errors.Wrap(err, "updating from remote TUF repository")
-			}
+		}
+		rootKeys, rootThreshold, err := getRootKeys(trustedRoot)
+		if err != nil {
+			return nil, errors.Wrap(err, "bad trusted root")
 		}
 
-		rootClient = uninitializedClient
+		if err := uninitializedClient.Init(rootKeys, rootThreshold); err != nil {
+			return nil, errors.Wrap(err, "initializing root client")
+		}
+		if err := updateMetadataAndDownloadTargets(uninitializedClient); err != nil {
+			return nil, errors.Wrap(err, "updating from remote TUF repository")
+		}
 	}
 
+	rootClient = uninitializedClient
 	return rootClient, nil
 }
 
+// Gets the global TUF client if the directory exists.
+// This will not make a remote call unless fetch is true.
+func RootClient(ctx context.Context, remote client.RemoteStore, altRoot []byte) (*client.Client, error) {
+	var err error
```

Comment:
This needs to be a global, or subsequent calls will return `nil, nil`

Reply:
fixed